WearComplicationProvidersTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/ComplicationToggleArgs.kt

```diff
@@ -27,11 +27,11 @@ import kotlinx.parcelize.Parcelize
 @Parcelize
 data class ComplicationToggleArgs(
     val providerComponent: ComponentName,
-    val complicationId: Int
+    val complicationInstanceId: Int
 ) : Parcelable
 
 /**
  * Returns the key for the preference used to hold the current state of a given complication.
  */
 fun ComplicationToggleArgs.getStatePreferenceKey(): Preferences.Key<Long> =
-    longPreferencesKey("${providerComponent.className}$complicationId")
+    longPreferencesKey("${providerComponent.className}$complicationInstanceId")
```

Comment:
This is unlikely to be stable across minified builds since R8 will obfuscate the class names differently, leading to the terrible user experience of preferences reset on each app update. Best to not show such an example to developers.

Reply:
Agreed, although you could consider turning obfuscation off for these classes.

Reply:
Then later on, someone enables minification again because R8 optimizations are wanted, or even needed (e.g. for Compose efficiency and jank-lessness), and it breaks.

IMO, there should be a lint rule with error level for these risky key naming schemes.

Reply:
In this instance, I believe this usage _should_ be fine since the class names are referring to `Service`s, which won't be obfuscated. These names have to be stable, since if they change, the complication will disappear from the watchface if the user was using it (very similar to widgets or wallpapers).

That being said, I totally agree that this usage is risky. I'll update it with a more straightfoward approach.

Reply:
Updated to use explicit keys in 8073a9a